[Automation] java run fails for net.bytebuddy:byte-buddy-agent:1.11.17

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.11.17" ./gradlew clean javaTest -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25481363170/job/74766432183
Last 300 lines of the log:
```console
BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: net.bytebuddy:byte-buddy-agent:1.10.9
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest > installCreatesTemporaryAgentJarFromInstallerClassResource() FAILED
    java.lang.IllegalStateException: The Byte Buddy agent is not initialized or unavailable
        at net.bytebuddy.agent.ByteBuddyAgent.doGetInstrumentation(ByteBuddyAgent.java:769)
        at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:607)
        at net_bytebuddy.byte_buddy_agent.ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.installCreatesTemporaryAgentJarFromInstallerClassResource(ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.java:48)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

8 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/tests/test/index.html


* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 7s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
15 actionable tasks: 7 executed, 8 up-to-date

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for net.bytebuddy:byte-buddy-agent:1.10.9 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 16s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.11.17 at 2026-05-07T07:11:58Z with exit code 1 after 16s
❌ ERROR: 'native-image run' for 1.11.17 failed with exit code 1.
SOFT-FAIL[native-image run][1.11.17][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.11.17" ./gradlew clean test -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"]
========================================================================================
 net.bytebuddy:byte-buddy-agent:1.10.9:1.11.17 stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.11.17 at 2026-05-07T07:11:58Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.11.17" ./gradlew clean compileTestJava -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 971ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.11.17 at 2026-05-07T07:12:01Z with exit code 0 after 3s
========================================================================================
 net.bytebuddy:byte-buddy-agent:1.10.9:1.11.17 stage java run
========================================================================================
Starting stage 'java run' for 1.11.17 at 2026-05-07T07:12:01Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.11.17" ./gradlew clean javaTest -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 840ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :tck-test-support:jar UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources
> Task :testClasses

> Task :test FAILED

ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest > installCreatesTemporaryAgentJarFromInstallerClassResource() FAILED
    java.lang.IllegalStateException: The Byte Buddy agent is not initialized or unavailable
        at net.bytebuddy.agent.ByteBuddyAgent.doGetInstrumentation(ByteBuddyAgent.java:769)
        at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:607)
        at net_bytebuddy.byte_buddy_agent.ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.installCreatesTemporaryAgentJarFromInstallerClassResource(ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.java:48)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

8 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 1.11.17 at 2026-05-07T07:12:06Z with exit code 1 after 5s
❌ ERROR: 'java run' for 1.11.17 failed with exit code 1.
FAILED[java run][1.11.17][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.11.17" ./gradlew clean javaTest -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.11.17" ./gradlew clean javaTest -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25481363170/job/74766432108
Last 300 lines of the log:
```console
BUILD SUCCESSFUL in 7s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: net.bytebuddy:byte-buddy-agent:1.10.9
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest > installCreatesTemporaryAgentJarFromInstallerClassResource() FAILED
    java.lang.IllegalStateException: The Byte Buddy agent is not initialized or unavailable
        at net.bytebuddy.agent.ByteBuddyAgent.doGetInstrumentation(ByteBuddyAgent.java:769)
        at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:607)
        at net_bytebuddy.byte_buddy_agent.ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.installCreatesTemporaryAgentJarFromInstallerClassResource(ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.java:48)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
8 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).



For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
15 actionable tasks: 7 executed, 8 up-to-date
BUILD FAILED in 7s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for net.bytebuddy:byte-buddy-agent:1.10.9 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 16s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.11.17 at 2026-05-07T07:11:53Z with exit code 1 after 16s
❌ ERROR: 'native-image run' for 1.11.17 failed with exit code 1.
SOFT-FAIL[native-image run][1.11.17][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.11.17" ./gradlew clean test -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"]
========================================================================================
 net.bytebuddy:byte-buddy-agent:1.10.9:1.11.17 stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.11.17 at 2026-05-07T07:11:53Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.11.17" ./gradlew clean compileTestJava -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 861ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 1s
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.11.17 at 2026-05-07T07:11:56Z with exit code 0 after 3s
========================================================================================
 net.bytebuddy:byte-buddy-agent:1.10.9:1.11.17 stage java run
========================================================================================
Starting stage 'java run' for 1.11.17 at 2026-05-07T07:11:56Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.11.17" ./gradlew clean javaTest -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 883ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources
> Task :testClasses

> Task :test FAILED

ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest > installCreatesTemporaryAgentJarFromInstallerClassResource() FAILED
    java.lang.IllegalStateException: The Byte Buddy agent is not initialized or unavailable
        at net.bytebuddy.agent.ByteBuddyAgent.doGetInstrumentation(ByteBuddyAgent.java:769)
        at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:607)
        at net_bytebuddy.byte_buddy_agent.ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.installCreatesTemporaryAgentJarFromInstallerClassResource(ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.java:48)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

8 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 1.11.17 at 2026-05-07T07:12:01Z with exit code 1 after 5s
❌ ERROR: 'java run' for 1.11.17 failed with exit code 1.
FAILED[java run][1.11.17][GVM_TCK_NATIVE_IMAGE_MODE="current-defaults" GVM_TCK_LV="1.11.17" ./gradlew clean javaTest -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.11.17" ./gradlew clean javaTest -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/25481363170/job/74766432235
Last 300 lines of the log:
```console
For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 5s
9 actionable tasks: 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :tckTest
====================
Testing library: net.bytebuddy:byte-buddy-agent:1.10.9
Command: `/home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/gradlew nativeTest`
Executing test...
-------
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :jar
> Task :checkTestTimeoutAnnotations
> Task :processTestResources
> Task :tck-test-support:compileJava
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes
> Task :tck-test-support:jar
> Task :compileTestJava
> Task :testClasses

> Task :test FAILED

ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest > installCreatesTemporaryAgentJarFromInstallerClassResource() FAILED
    java.lang.IllegalStateException: The Byte Buddy agent is not initialized or unavailable
        at net.bytebuddy.agent.ByteBuddyAgent.doGetInstrumentation(ByteBuddyAgent.java:769)
        at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:607)
        at net_bytebuddy.byte_buddy_agent.ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.installCreatesTemporaryAgentJarFromInstallerClassResource(ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.java:48)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
15 actionable tasks: 7 executed, 8 up-to-date

8 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 5s

> Task :tckTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':tckTest'.
> Test for net.bytebuddy:byte-buddy-agent:1.10.9 failed with exit code 1.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 12s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'native-image run' for 1.11.17 at 2026-05-07T07:12:34Z with exit code 1 after 13s
❌ ERROR: 'native-image run' for 1.11.17 failed with exit code 1.
SOFT-FAIL[native-image run][1.11.17][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.11.17" ./gradlew clean test -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"]
========================================================================================
 net.bytebuddy:byte-buddy-agent:1.10.9:1.11.17 stage javac compile
========================================================================================
Starting stage 'javac compile' for 1.11.17 at 2026-05-07T07:12:34Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.11.17" ./gradlew clean compileTestJava -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 699ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :compileTestJava
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :compileJava NO-SOURCE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :compileTestJava

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 921ms
10 actionable tasks: 1 executed, 9 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 2s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'javac compile' for 1.11.17 at 2026-05-07T07:12:36Z with exit code 0 after 2s
========================================================================================
 net.bytebuddy:byte-buddy-agent:1.10.9:1.11.17 stage java run
========================================================================================
Starting stage 'java run' for 1.11.17 at 2026-05-07T07:12:36Z
Command: GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.11.17" ./gradlew clean javaTest -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE

> Configure project :
GraalVM Reachability Metadata TCK
---------------------------------

> Task :tckClean
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :clean

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD SUCCESSFUL in 642ms
9 actionable tasks: 1 executed, 8 up-to-date
Consider enabling configuration cache to speed up this build: https://docs.gradle.org/9.1.0/userguide/configuration_cache_enabling.html

> Task :clean

> Task :javaTest FAILED
> Task :tck-build-logic:extractPluginRequests UP-TO-DATE
> Task :tck-build-logic:generatePluginAdapters UP-TO-DATE
> Task :tck-build-logic:compileJava UP-TO-DATE
> Task :tck-build-logic:compileGroovy UP-TO-DATE
> Task :tck-build-logic:compileGroovyPlugins UP-TO-DATE
> Task :tck-build-logic:pluginDescriptors UP-TO-DATE
> Task :tck-build-logic:processResources UP-TO-DATE
> Task :tck-build-logic:classes UP-TO-DATE
> Task :tck-build-logic:jar UP-TO-DATE
> Task :checkTestTimeoutAnnotations
> Task :compileJava NO-SOURCE
> Task :tck-test-support:compileJava UP-TO-DATE
> Task :processResources NO-SOURCE
> Task :classes UP-TO-DATE
> Task :tck-test-support:processResources NO-SOURCE
> Task :tck-test-support:classes UP-TO-DATE
> Task :tck-test-support:jar UP-TO-DATE
> Task :compileTestJava
> Task :processTestResources
> Task :testClasses

> Task :test FAILED

ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest > installCreatesTemporaryAgentJarFromInstallerClassResource() FAILED
    java.lang.IllegalStateException: The Byte Buddy agent is not initialized or unavailable
        at net.bytebuddy.agent.ByteBuddyAgent.doGetInstrumentation(ByteBuddyAgent.java:769)
        at net.bytebuddy.agent.ByteBuddyAgent.install(ByteBuddyAgent.java:607)
        at net_bytebuddy.byte_buddy_agent.ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.installCreatesTemporaryAgentJarFromInstallerClassResource(ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.java:48)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
14 actionable tasks: 4 executed, 10 up-to-date

8 tests completed, 1 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/net.bytebuddy/byte-buddy-agent/1.10.9/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 1s

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 3s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 1.11.17 at 2026-05-07T07:12:40Z with exit code 1 after 4s
❌ ERROR: 'java run' for 1.11.17 failed with exit code 1.
FAILED[java run][1.11.17][GVM_TCK_NATIVE_IMAGE_MODE="future-defaults-all" GVM_TCK_LV="1.11.17" ./gradlew clean javaTest -Pcoordinates="net.bytebuddy:byte-buddy-agent:1.10.9"]
```

--- a/metadata/net.bytebuddy/byte-buddy-agent/1.11.17/reachability-metadata.json
+++ b/metadata/net.bytebuddy/byte-buddy-agent/1.11.17/reachability-metadata.json
@@ -1,86 +1,74 @@
 {
-  "reflection": [
+  "reflection" : [
     {
-      "condition": {
-        "typeReached": "net.bytebuddy.agent.ByteBuddyAgent"
+      "condition" : {
+        "typeReached" : "net.bytebuddy.agent.ByteBuddyAgent"
       },
-      "type": "byte[][]"
+      "type" : "byte[][]"
     },
     {
-      "condition": {
-        "typeReached": "net.bytebuddy.agent.ByteBuddyAgent$AgentProvider$ForByteBuddyAgent"
+      "condition" : {
+        "typeReached" : "net.bytebuddy.agent.ByteBuddyAgent$AgentProvider$ForByteBuddyAgent"
       },
-      "type": "java.lang.StackTraceElement[]"
+      "type" : "java.lang.StackTraceElement[]"
     },
     {
-      "condition": {
-        "typeReached": "net.bytebuddy.agent.Attacher"
+      "condition" : {
+        "typeReached" : "net.bytebuddy.agent.Attacher"
       },
-      "type": "java.lang.reflect.Method[]"
+      "type" : "java.lang.reflect.Method[]"
     },
     {
-      "condition": {
-        "typeReached": "net.bytebuddy.agent.Installer"
+      "condition" : {
+        "typeReached" : "net.bytebuddy.agent.Installer"
       },
-      "type": "java.lang.reflect.Method[]"
+      "type" : "java.lang.reflect.Method[]"
     },
     {
-      "condition": {
-        "typeReached": "net.bytebuddy.agent.Attacher"
+      "condition" : {
+        "typeReached" : "net.bytebuddy.agent.ByteBuddyAgent$AgentProvider$ForByteBuddyAgent"
       },
-      "type": "net_bytebuddy.byte_buddy_agent.AttacherTest$RecordingVirtualMachine"
+      "type" : "sun.security.provider.NativePRNG"
     },
     {
-      "condition": {
-        "typeReached": "net.bytebuddy.agent.Attacher"
+      "condition" : {
+        "typeReached" : "net.bytebuddy.agent.ByteBuddyAgent$AgentProvider$ForByteBuddyAgent"
       },
-      "type": "net_bytebuddy.byte_buddy_agent.ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest$RecordingVirtualMachine"
+      "type" : "sun.security.provider.SHA"
     },
     {
-      "condition": {
-        "typeReached": "net.bytebuddy.agent.ByteBuddyAgent$AgentProvider$ForByteBuddyAgent"
+      "condition" : {
+        "typeReached" : "net.bytebuddy.agent.Installer"
       },
-      "type": "sun.security.provider.NativePRNG"
-    },
-    {
-      "condition": {
-        "typeReached": "net.bytebuddy.agent.ByteBuddyAgent$AgentProvider$ForByteBuddyAgent"
-      },
-      "type": "sun.security.provider.SHA"
-    },
-    {
-      "condition": {
-        "typeReached": "net.bytebuddy.agent.Installer"
-      },
-      "type": {
-        "proxy": [
+      "type" : {
+        "proxy" : [
           "java.lang.Deprecated"
         ]
       }
     },
     {
-      "condition": {
-        "typeReached": "net.bytebuddy.agent.Installer"
+      "condition" : {
+        "typeReached" : "net.bytebuddy.agent.Installer"
       },
-      "type": {
-        "proxy": [
+      "type" : {
+        "proxy" : [
           "java.lang.annotation.Retention"
         ]
       }
     }
   ],
-  "resources": [
+  "resources" : [
     {
-      "condition": {
-        "typeReached": "net.bytebuddy.agent.ByteBuddyAgent"
+      "condition" : {
+        "typeReached" : "net.bytebuddy.agent.ByteBuddyAgent"
       },
-      "glob": "net/bytebuddy/agent/Attacher.class"
+      "glob" : "net/bytebuddy/agent/Attacher.class"
     },
     {
-      "condition": {
-        "typeReached": "net.bytebuddy.agent.ByteBuddyAgent$AgentProvider$ForByteBuddyAgent"
+      "condition" : {
+        "typeReached" : "net.bytebuddy.agent.ByteBuddyAgent$AgentProvider$ForByteBuddyAgent"
       },
-      "glob": "net/bytebuddy/agent/Installer.class"
+      "glob" : "net/bytebuddy/agent/Installer.class"
     }
   ]
 }

--- a/metadata/net.bytebuddy/byte-buddy-agent/1.11.17/reachability-metadata.json
+++ b/metadata/net.bytebuddy/byte-buddy-agent/1.11.17/reachability-metadata.json
@@ -1,0 +1,86 @@
+{
+  "reflection": [
+    {
+      "condition": {
+        "typeReached": "net.bytebuddy.agent.ByteBuddyAgent"
+      },
+      "type": "byte[][]"
+    },
+    {
+      "condition": {
+        "typeReached": "net.bytebuddy.agent.ByteBuddyAgent$AgentProvider$ForByteBuddyAgent"
+      },
+      "type": "java.lang.StackTraceElement[]"
+    },
+    {
+      "condition": {
+        "typeReached": "net.bytebuddy.agent.Attacher"
+      },
+      "type": "java.lang.reflect.Method[]"
+    },
+    {
+      "condition": {
+        "typeReached": "net.bytebuddy.agent.Installer"
+      },
+      "type": "java.lang.reflect.Method[]"
+    },
+    {
+      "condition": {
+        "typeReached": "net.bytebuddy.agent.Attacher"
+      },
+      "type": "net_bytebuddy.byte_buddy_agent.AttacherTest$RecordingVirtualMachine"
+    },
+    {
+      "condition": {
+        "typeReached": "net.bytebuddy.agent.Attacher"
+      },
+      "type": "net_bytebuddy.byte_buddy_agent.ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest$RecordingVirtualMachine"
+    },
+    {
+      "condition": {
+        "typeReached": "net.bytebuddy.agent.ByteBuddyAgent$AgentProvider$ForByteBuddyAgent"
+      },
+      "type": "sun.security.provider.NativePRNG"
+    },
+    {
+      "condition": {
+        "typeReached": "net.bytebuddy.agent.ByteBuddyAgent$AgentProvider$ForByteBuddyAgent"
+      },
+      "type": "sun.security.provider.SHA"
+    },
+    {
+      "condition": {
+        "typeReached": "net.bytebuddy.agent.Installer"
+      },
+      "type": {
+        "proxy": [
+          "java.lang.Deprecated"
+        ]
+      }
+    },
+    {
+      "condition": {
+        "typeReached": "net.bytebuddy.agent.Installer"
+      },
+      "type": {
+        "proxy": [
+          "java.lang.annotation.Retention"
+        ]
+      }
+    }
+  ],
+  "resources": [
+    {
+      "condition": {
+        "typeReached": "net.bytebuddy.agent.ByteBuddyAgent"
+      },
+      "glob": "net/bytebuddy/agent/Attacher.class"
+    },
+    {
+      "condition": {
+        "typeReached": "net.bytebuddy.agent.ByteBuddyAgent$AgentProvider$ForByteBuddyAgent"
+      },
+      "glob": "net/bytebuddy/agent/Installer.class"
+    }
+  ]
+}

--- a/metadata/net.bytebuddy/byte-buddy-agent/index.json
+++ b/metadata/net.bytebuddy/byte-buddy-agent/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "1.11.17",
+    "tested-versions" : [
+      "1.11.17"
+    ],
+    "allowed-packages" : [
+      "net.bytebuddy.agent"
+    ]
+  },
+  {
     "metadata-version" : "1.10.9",
     "source-code-url" : "https://repo.maven.apache.org/maven2/net/bytebuddy/byte-buddy-agent/$version$/byte-buddy-agent-$version$-sources.jar",
     "repository-url" : "https://github.com/raphw/byte-buddy",

--- a/metadata/net.bytebuddy/byte-buddy-agent/index.json
+++ b/metadata/net.bytebuddy/byte-buddy-agent/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "1.11.17",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/net/bytebuddy/byte-buddy-agent/$version$/byte-buddy-agent-$version$-sources.jar",
+    "repository-url" : "https://github.com/raphw/byte-buddy",
+    "test-code-url" : "https://github.com/raphw/byte-buddy/tree/byte-buddy-$version$/byte-buddy-agent/src/test/java",
+    "documentation-url" : "https://repo.maven.apache.org/maven2/net/bytebuddy/byte-buddy-agent/$version$/byte-buddy-agent-$version$-javadoc.jar",
+    "description" : "Byte Buddy Agent provides APIs and a Java agent for attaching instrumentation to the current JVM or to another running JVM. It is used to install Java instrumentation at runtime so applications and frameworks can redefine, rebase, or transform classes through Byte Buddy.",
     "tested-versions" : [
       "1.11.17"
     ],

--- a/stats/net.bytebuddy/byte-buddy-agent/1.11.17/execution-metrics.json
+++ b/stats/net.bytebuddy/byte-buddy-agent/1.11.17/execution-metrics.json
@@ -1,0 +1,111 @@
+{
+  "fix_java_run_fail:2026-06-26": {
+    "timestamp": "2026-06-26T13:48:29.170861Z",
+    "library": "net.bytebuddy:byte-buddy-agent:1.11.17",
+    "starting_commit": "ef8e30942bfd0442fd44b59a204a6ba2e6cb6afc",
+    "ending_commit": "93632c0d078f082f9fed1a1c478c3f6b95bd0c05",
+    "previous_library": "net.bytebuddy:byte-buddy-agent:1.10.9",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "1.11.17",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 15,
+            "totalCalls": 15
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 17,
+        "totalCalls": 17
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 821,
+          "missed": 4044,
+          "ratio": 0.168756,
+          "total": 4865
+        },
+        "line": {
+          "covered": 167,
+          "missed": 823,
+          "ratio": 0.168687,
+          "total": 990
+        },
+        "method": {
+          "covered": 42,
+          "missed": 149,
+          "ratio": 0.219895,
+          "total": 191
+        }
+      }
+    },
+    "previous_library_stats": {
+      "version": "1.10.9",
+      "dynamicAccess": {
+        "breakdown": {
+          "reflection": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 14,
+            "totalCalls": 14
+          },
+          "resources": {
+            "coverageRatio": 1.0,
+            "coveredCalls": 2,
+            "totalCalls": 2
+          }
+        },
+        "coverageRatio": 1.0,
+        "coveredCalls": 16,
+        "totalCalls": 16
+      },
+      "libraryCoverage": {
+        "instruction": {
+          "covered": 761,
+          "missed": 3990,
+          "ratio": 0.160177,
+          "total": 4751
+        },
+        "line": {
+          "covered": 158,
+          "missed": 808,
+          "ratio": 0.163561,
+          "total": 966
+        },
+        "method": {
+          "covered": 40,
+          "missed": 150,
+          "ratio": 0.210526,
+          "total": 190
+        }
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 111423,
+      "cached_input_tokens_used": 915456,
+      "output_tokens_used": 13809,
+      "iterations": 4,
+      "cost_usd": 0.872,
+      "tested_library_loc": 167,
+      "metadata_entries": 10,
+      "test_only_metadata_entries": 2,
+      "previous_library_metadata_entries": 11,
+      "previous_library_test_only_metadata_entries": 3,
+      "code_coverage_percent": 16.87,
+      "previous_library_coverage_percent": 16.36
+    },
+    "artifacts": {
+      "test_file": "tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/com/ibm/tools/attach/VirtualMachine.java",
+      "metadata_file": "metadata/net.bytebuddy/byte-buddy-agent/1.11.17/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/net.bytebuddy/byte-buddy-agent/1.11.17/stats.json
+++ b/stats/net.bytebuddy/byte-buddy-agent/1.11.17/stats.json
@@ -1,0 +1,42 @@
+{
+  "versions" : [ {
+    "version" : "1.11.17",
+    "dynamicAccess" : {
+      "breakdown" : {
+        "reflection" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 15,
+          "totalCalls" : 15
+        },
+        "resources" : {
+          "coverageRatio" : 1.0,
+          "coveredCalls" : 2,
+          "totalCalls" : 2
+        }
+      },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 17,
+      "totalCalls" : 17
+    },
+    "libraryCoverage" : {
+      "instruction" : {
+        "covered" : 821,
+        "missed" : 4044,
+        "ratio" : 0.168756,
+        "total" : 4865
+      },
+      "line" : {
+        "covered" : 167,
+        "missed" : 823,
+        "ratio" : 0.168687,
+        "total" : 990
+      },
+      "method" : {
+        "covered" : 42,
+        "missed" : 149,
+        "ratio" : 0.219895,
+        "total" : 191
+      }
+    }
+  } ]
+}

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/.gitignore
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/build.gradle
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "net.bytebuddy:byte-buddy-agent:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/build.gradle
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/build.gradle
@@ -15,6 +15,25 @@ dependencies {
     testImplementation 'org.assertj:assertj-core:3.22.0'
 }
 
+java {
+    toolchain {
+        languageVersion = JavaLanguageVersion.of(21)
+    }
+}
+
+tasks.named("test") {
+    javaLauncher = javaToolchains.launcherFor {
+        languageVersion = JavaLanguageVersion.of(21)
+    }
+    jvmArgs += ["-Djava.security.manager=allow"]
+}
+
+tasks.named("nativeTest") {
+    if ((JavaVersion.current().majorVersion as int) < 24) {
+        runtimeArgs.add("-Djava.security.manager=allow")
+    }
+}
+
 graalvmNative {
     agent {
         defaultMode = "conditional"

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/gradle.properties
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = net.bytebuddy:byte-buddy-agent:1.11.17
+library.version = 1.11.17
+metadata.dir = net.bytebuddy/byte-buddy-agent/1.11.17/

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/settings.gradle
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'net.bytebuddy.byte-buddy-agent_tests'

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/com/ibm/tools/attach/VirtualMachine.java
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/com/ibm/tools/attach/VirtualMachine.java
@@ -1,0 +1,12 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package com.ibm.tools.attach;
+
+public final class VirtualMachine {
+    private VirtualMachine() {
+    }
+}

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/AttacherTest.java
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/AttacherTest.java
@@ -1,0 +1,93 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package net_bytebuddy.byte_buddy_agent;
+
+import net.bytebuddy.agent.Attacher;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AttacherTest {
+    @BeforeEach
+    void resetVirtualMachine() {
+        RecordingVirtualMachine.reset();
+    }
+
+    @Test
+    void mainLoadsJavaAgentAndCombinesArguments() {
+        Attacher.main(new String[] {
+                RecordingVirtualMachine.class.getName(),
+                "1234",
+                "agent.jar",
+                "false",
+                "=first",
+                "second",
+                "third"
+        });
+
+        assertThat(RecordingVirtualMachine.attachedProcessId).isEqualTo("1234");
+        assertThat(RecordingVirtualMachine.loadedAgent).isEqualTo("agent.jar");
+        assertThat(RecordingVirtualMachine.loadedAgentArgument).isEqualTo("first second third");
+        assertThat(RecordingVirtualMachine.loadedNativeAgent).isNull();
+        assertThat(RecordingVirtualMachine.detached).isTrue();
+    }
+
+    @Test
+    void mainLoadsNativeAgentWithNullArgument() {
+        Attacher.main(new String[] {
+                RecordingVirtualMachine.class.getName(),
+                "5678",
+                "native-agent.so",
+                "true",
+                ""
+        });
+
+        assertThat(RecordingVirtualMachine.attachedProcessId).isEqualTo("5678");
+        assertThat(RecordingVirtualMachine.loadedNativeAgent).isEqualTo("native-agent.so");
+        assertThat(RecordingVirtualMachine.loadedNativeAgentArgument).isNull();
+        assertThat(RecordingVirtualMachine.loadedAgent).isNull();
+        assertThat(RecordingVirtualMachine.detached).isTrue();
+    }
+
+    public static class RecordingVirtualMachine {
+        private static String attachedProcessId;
+        private static String loadedAgent;
+        private static String loadedAgentArgument;
+        private static String loadedNativeAgent;
+        private static String loadedNativeAgentArgument;
+        private static boolean detached;
+
+        public static RecordingVirtualMachine attach(String processId) {
+            attachedProcessId = processId;
+            return new RecordingVirtualMachine();
+        }
+
+        public void loadAgent(String agent, String argument) {
+            loadedAgent = agent;
+            loadedAgentArgument = argument;
+        }
+
+        public void loadAgentPath(String agent, String argument) {
+            loadedNativeAgent = agent;
+            loadedNativeAgentArgument = argument;
+        }
+
+        public void detach() {
+            detached = true;
+        }
+
+        private static void reset() {
+            attachedProcessId = null;
+            loadedAgent = null;
+            loadedAgentArgument = null;
+            loadedNativeAgent = null;
+            loadedNativeAgentArgument = null;
+            detached = false;
+        }
+    }
+}

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.java
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.java
@@ -1,0 +1,120 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package net_bytebuddy.byte_buddy_agent;
+
+import net.bytebuddy.agent.ByteBuddyAgent;
+import net.bytebuddy.agent.Installer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.lang.instrument.Instrumentation;
+import java.util.Collections;
+import java.util.jar.JarFile;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest {
+    private static final String PROCESS_ID = "4711";
+
+    private String originalLatentResolve;
+
+    @BeforeEach
+    void resetState() {
+        originalLatentResolve = System.getProperty(ByteBuddyAgent.LATENT_RESOLVE);
+        System.setProperty(ByteBuddyAgent.LATENT_RESOLVE, Boolean.TRUE.toString());
+        Installer.premain(null, null);
+        RecordingVirtualMachine.reset();
+    }
+
+    @AfterEach
+    void restoreState() {
+        Installer.premain(null, null);
+        RecordingVirtualMachine.reset();
+        if (originalLatentResolve == null) {
+            System.clearProperty(ByteBuddyAgent.LATENT_RESOLVE);
+        } else {
+            System.setProperty(ByteBuddyAgent.LATENT_RESOLVE, originalLatentResolve);
+        }
+    }
+
+    @Test
+    void installCreatesTemporaryAgentJarFromInstallerClassResource() throws Exception {
+        Instrumentation instrumentation = ByteBuddyAgent.install(new RecordingAttachmentProvider(), () -> PROCESS_ID);
+
+        assertThat(instrumentation).isNull();
+        assertThat(RecordingVirtualMachine.attachedProcessId).isEqualTo(PROCESS_ID);
+        assertThat(RecordingVirtualMachine.loadedAgentArgument).isNull();
+        assertThat(RecordingVirtualMachine.detached).isTrue();
+
+        File agentJar = new File(RecordingVirtualMachine.loadedAgent);
+        assertThat(agentJar).isFile();
+        try (JarFile jarFile = new JarFile(agentJar)) {
+            assertThat(jarFile.getEntry(Installer.class.getName().replace('.', '/') + ".class")).isNotNull();
+            assertThat(jarFile.getManifest().getMainAttributes().getValue("Agent-Class"))
+                    .isEqualTo(Installer.class.getName());
+        }
+    }
+
+    private static final class RecordingAttachmentProvider implements ByteBuddyAgent.AttachmentProvider {
+        @Override
+        public ByteBuddyAgent.AttachmentProvider.Accessor attempt() {
+            return new RecordingAccessor();
+        }
+    }
+
+    private static final class RecordingAccessor implements ByteBuddyAgent.AttachmentProvider.Accessor {
+        @Override
+        public boolean isAvailable() {
+            return true;
+        }
+
+        @Override
+        public boolean isExternalAttachmentRequired() {
+            return false;
+        }
+
+        @Override
+        public Class<?> getVirtualMachineType() {
+            return RecordingVirtualMachine.class;
+        }
+
+        @Override
+        public ExternalAttachment getExternalAttachment() {
+            return new ExternalAttachment(RecordingVirtualMachine.class.getName(), Collections.<File>emptyList());
+        }
+    }
+
+    public static final class RecordingVirtualMachine {
+        private static String attachedProcessId;
+        private static String loadedAgent;
+        private static String loadedAgentArgument;
+        private static boolean detached;
+
+        public static RecordingVirtualMachine attach(String processId) {
+            attachedProcessId = processId;
+            return new RecordingVirtualMachine();
+        }
+
+        public void loadAgent(String agent, String argument) {
+            loadedAgent = agent;
+            loadedAgentArgument = argument;
+        }
+
+        public void detach() {
+            detached = true;
+        }
+
+        private static void reset() {
+            attachedProcessId = null;
+            loadedAgent = null;
+            loadedAgentArgument = null;
+            detached = false;
+        }
+    }
+}

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.java
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest.java
@@ -13,8 +13,14 @@ import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 import java.io.File;
+import java.lang.instrument.ClassDefinition;
+import java.lang.instrument.ClassFileTransformer;
 import java.lang.instrument.Instrumentation;
+import java.lang.instrument.UnmodifiableClassException;
 import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
 import java.util.jar.JarFile;
 
 import static org.assertj.core.api.Assertions.assertThat;
@@ -47,7 +53,7 @@ public class ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest {
     void installCreatesTemporaryAgentJarFromInstallerClassResource() throws Exception {
         Instrumentation instrumentation = ByteBuddyAgent.install(new RecordingAttachmentProvider(), () -> PROCESS_ID);
 
-        assertThat(instrumentation).isNull();
+        assertThat(instrumentation).isSameAs(RecordingVirtualMachine.instrumentation);
         assertThat(RecordingVirtualMachine.attachedProcessId).isEqualTo(PROCESS_ID);
         assertThat(RecordingVirtualMachine.loadedAgentArgument).isNull();
         assertThat(RecordingVirtualMachine.detached).isTrue();
@@ -95,6 +101,7 @@ public class ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest {
         private static String loadedAgent;
         private static String loadedAgentArgument;
         private static boolean detached;
+        private static Instrumentation instrumentation;
 
         public static RecordingVirtualMachine attach(String processId) {
             attachedProcessId = processId;
@@ -104,6 +111,8 @@ public class ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest {
         public void loadAgent(String agent, String argument) {
             loadedAgent = agent;
             loadedAgentArgument = argument;
+            instrumentation = new RecordingInstrumentation();
+            Installer.agentmain(argument, instrumentation);
         }
 
         public void detach() {
@@ -115,6 +124,92 @@ public class ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest {
             loadedAgent = null;
             loadedAgentArgument = null;
             detached = false;
+            instrumentation = null;
+        }
+    }
+
+    private static final class RecordingInstrumentation implements Instrumentation {
+        @Override
+        public void addTransformer(ClassFileTransformer transformer, boolean canRetransform) {
+        }
+
+        @Override
+        public void addTransformer(ClassFileTransformer transformer) {
+        }
+
+        @Override
+        public boolean removeTransformer(ClassFileTransformer transformer) {
+            return false;
+        }
+
+        @Override
+        public boolean isRetransformClassesSupported() {
+            return false;
+        }
+
+        @Override
+        public void retransformClasses(Class<?>... classes) throws UnmodifiableClassException {
+        }
+
+        @Override
+        public boolean isRedefineClassesSupported() {
+            return false;
+        }
+
+        @Override
+        public void redefineClasses(ClassDefinition... definitions)
+                throws ClassNotFoundException, UnmodifiableClassException {
+        }
+
+        @Override
+        public boolean isModifiableClass(Class<?> type) {
+            return false;
+        }
+
+        @Override
+        public Class<?>[] getAllLoadedClasses() {
+            return new Class<?>[0];
+        }
+
+        @Override
+        public Class<?>[] getInitiatedClasses(ClassLoader loader) {
+            return new Class<?>[0];
+        }
+
+        @Override
+        public long getObjectSize(Object object) {
+            return 0L;
+        }
+
+        @Override
+        public void appendToBootstrapClassLoaderSearch(JarFile jarFile) {
+        }
+
+        @Override
+        public void appendToSystemClassLoaderSearch(JarFile jarFile) {
+        }
+
+        @Override
+        public boolean isNativeMethodPrefixSupported() {
+            return false;
+        }
+
+        @Override
+        public void setNativeMethodPrefix(ClassFileTransformer transformer, String prefix) {
+        }
+
+        @Override
+        public void redefineModule(Module module,
+                                   Set<Module> extraReads,
+                                   Map<String, Set<Module>> extraExports,
+                                   Map<String, Set<Module>> extraOpens,
+                                   Set<Class<?>> extraUses,
+                                   Map<Class<?>, List<Class<?>>> extraProvides) {
+        }
+
+        @Override
+        public boolean isModifiableModule(Module module) {
+            return false;
         }
     }
 }

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/ByteBuddyAgentInnerAttachmentProviderInnerAccessorInnerSimpleTest.java
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/ByteBuddyAgentInnerAttachmentProviderInnerAccessorInnerSimpleTest.java
@@ -1,0 +1,70 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package net_bytebuddy.byte_buddy_agent;
+
+import com.ibm.tools.attach.VirtualMachine;
+import net.bytebuddy.agent.ByteBuddyAgent;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.File;
+import java.nio.file.Path;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ByteBuddyAgentInnerAttachmentProviderInnerAccessorInnerSimpleTest {
+    @Test
+    void createsExternalAttachmentAccessorFromProvidedClassLoader(@TempDir Path temporaryDirectory) {
+        File toolsJar = temporaryDirectory.resolve("tools.jar").toFile();
+        RecordingClassLoader classLoader = new RecordingClassLoader();
+
+        ByteBuddyAgent.AttachmentProvider.Accessor accessor =
+                ByteBuddyAgent.AttachmentProvider.Accessor.Simple.of(classLoader, toolsJar);
+
+        assertThat(classLoader.requestedName)
+                .isEqualTo(ByteBuddyAgent.AttachmentProvider.Accessor.VIRTUAL_MACHINE_TYPE_NAME);
+        assertThat(accessor.isAvailable()).isTrue();
+        assertThat(accessor.isExternalAttachmentRequired()).isTrue();
+        assertThat(accessor.getVirtualMachineType()).isEqualTo(RecordingVirtualMachine.class);
+        assertThat(accessor.getExternalAttachment().getVirtualMachineType())
+                .isEqualTo(RecordingVirtualMachine.class.getName());
+        assertThat(accessor.getExternalAttachment().getClassPath()).containsExactly(toolsJar);
+    }
+
+    @Test
+    void createsExternalAttachmentAccessorForJ9VirtualMachineClassFromSystemClassLoader() {
+        ByteBuddyAgent.AttachmentProvider.Accessor accessor =
+                ByteBuddyAgent.AttachmentProvider.Accessor.Simple.ofJ9();
+
+        assertThat(accessor.isAvailable()).isTrue();
+        assertThat(accessor.isExternalAttachmentRequired()).isTrue();
+        assertThat(accessor.getVirtualMachineType()).isEqualTo(VirtualMachine.class);
+        assertThat(accessor.getExternalAttachment().getVirtualMachineType())
+                .isEqualTo(ByteBuddyAgent.AttachmentProvider.Accessor.VIRTUAL_MACHINE_TYPE_NAME_J9);
+        assertThat(accessor.getExternalAttachment().getClassPath()).isEmpty();
+    }
+
+    private static final class RecordingClassLoader extends ClassLoader {
+        private String requestedName;
+
+        private RecordingClassLoader() {
+            super(null);
+        }
+
+        @Override
+        public Class<?> loadClass(String name) throws ClassNotFoundException {
+            requestedName = name;
+            if (ByteBuddyAgent.AttachmentProvider.Accessor.VIRTUAL_MACHINE_TYPE_NAME.equals(name)) {
+                return RecordingVirtualMachine.class;
+            }
+            throw new ClassNotFoundException(name);
+        }
+    }
+
+    public static final class RecordingVirtualMachine {
+    }
+}

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/ByteBuddyAgentInnerAttachmentProviderInnerAccessorInnerSimpleTest.java
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/ByteBuddyAgentInnerAttachmentProviderInnerAccessorInnerSimpleTest.java
@@ -40,7 +40,10 @@ public class ByteBuddyAgentInnerAttachmentProviderInnerAccessorInnerSimpleTest {
         ByteBuddyAgent.AttachmentProvider.Accessor accessor =
                 ByteBuddyAgent.AttachmentProvider.Accessor.Simple.ofJ9();
 
-        assertThat(accessor.isAvailable()).isTrue();
+        assertThat(accessor.isAvailable()).isEqualTo(isJ9VirtualMachineAvailableFromSystemClassLoader());
+        if (!accessor.isAvailable()) {
+            return;
+        }
         assertThat(accessor.isExternalAttachmentRequired()).isTrue();
         assertThat(accessor.getVirtualMachineType()).isEqualTo(VirtualMachine.class);
         assertThat(accessor.getExternalAttachment().getVirtualMachineType())
@@ -66,5 +69,15 @@ public class ByteBuddyAgentInnerAttachmentProviderInnerAccessorInnerSimpleTest {
     }
 
     public static final class RecordingVirtualMachine {
+    }
+
+    private static boolean isJ9VirtualMachineAvailableFromSystemClassLoader() {
+        try {
+            return ClassLoader.getSystemClassLoader()
+                    .loadClass(ByteBuddyAgent.AttachmentProvider.Accessor.VIRTUAL_MACHINE_TYPE_NAME_J9)
+                    == VirtualMachine.class;
+        } catch (ClassNotFoundException ignored) {
+            return false;
+        }
     }
 }

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/ByteBuddyAgentTest.java
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/ByteBuddyAgentTest.java
@@ -1,0 +1,233 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package net_bytebuddy.byte_buddy_agent;
+
+import net.bytebuddy.agent.ByteBuddyAgent;
+import net.bytebuddy.agent.Installer;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.instrument.ClassDefinition;
+import java.lang.instrument.ClassFileTransformer;
+import java.lang.instrument.Instrumentation;
+import java.lang.instrument.UnmodifiableClassException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.jar.JarFile;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class ByteBuddyAgentTest {
+    private static final String ALLOW_SELF_ATTACH = "jdk.attach.allowAttachSelf";
+    private static final String JAVA_HOME = "java.home";
+
+    private static String originalAllowSelfAttach;
+
+    @BeforeAll
+    static void requireExternalSelfAttachment() {
+        originalAllowSelfAttach = System.getProperty(ALLOW_SELF_ATTACH);
+        System.setProperty(ALLOW_SELF_ATTACH, Boolean.FALSE.toString());
+    }
+
+    @AfterAll
+    static void restoreSelfAttachmentProperty() {
+        restoreProperty(ALLOW_SELF_ATTACH, originalAllowSelfAttach);
+    }
+
+    @BeforeEach
+    void resetInstaller() {
+        Installer.premain(null, null);
+    }
+
+    @Test
+    void getInstrumentationLooksUpInstallerViaSystemClassLoader() {
+        assertThatThrownBy(ByteBuddyAgent::getInstrumentation)
+                .isInstanceOf(IllegalStateException.class)
+                .hasMessageContaining("not initialized");
+    }
+
+    @Test
+    void getInstrumentationInvokesInstallerMethodSuccessfully() {
+        Instrumentation instrumentation = new RecordingInstrumentation();
+
+        Installer.premain(null, instrumentation);
+
+        assertThat(ByteBuddyAgent.getInstrumentation()).isSameAs(instrumentation);
+    }
+
+    @Test
+    void externalAttachmentCopiesAttacherClassResource(@TempDir Path temporaryDirectory) throws IOException {
+        Path agentJar = Files.createFile(temporaryDirectory.resolve("agent.jar"));
+        String originalLatentResolve = System.getProperty(ByteBuddyAgent.LATENT_RESOLVE);
+        String originalJavaHome = System.getProperty(JAVA_HOME);
+        System.setProperty(ByteBuddyAgent.LATENT_RESOLVE, Boolean.TRUE.toString());
+        System.setProperty(JAVA_HOME, temporaryDirectory.resolve("missing-java-home").toString());
+        try {
+            assertThatThrownBy(() -> ByteBuddyAgent.attach(agentJar.toFile(),
+                    ByteBuddyAgent.ProcessProvider.ForCurrentVm.INSTANCE,
+                    new AlwaysExternalAttachmentProvider()))
+                    .isInstanceOf(IllegalStateException.class)
+                    .hasMessageContaining("Error during attachment");
+        } finally {
+            restoreProperty(ByteBuddyAgent.LATENT_RESOLVE, originalLatentResolve);
+            restoreProperty(JAVA_HOME, originalJavaHome);
+        }
+    }
+
+    private static void restoreProperty(String name, String value) {
+        if (value == null) {
+            System.clearProperty(name);
+        } else {
+            System.setProperty(name, value);
+        }
+    }
+
+    private static final class AlwaysExternalAttachmentProvider implements ByteBuddyAgent.AttachmentProvider {
+        @Override
+        public ByteBuddyAgent.AttachmentProvider.Accessor attempt() {
+            return new AlwaysExternalAccessor();
+        }
+    }
+
+    private static final class AlwaysExternalAccessor implements ByteBuddyAgent.AttachmentProvider.Accessor {
+        @Override
+        public boolean isAvailable() {
+            return true;
+        }
+
+        @Override
+        public boolean isExternalAttachmentRequired() {
+            return true;
+        }
+
+        @Override
+        public Class<?> getVirtualMachineType() {
+            return RecordingVirtualMachine.class;
+        }
+
+        @Override
+        public ExternalAttachment getExternalAttachment() {
+            return new ExternalAttachment(RecordingVirtualMachine.class.getName(), Collections.<File>emptyList());
+        }
+    }
+
+    private static final class RecordingInstrumentation implements Instrumentation {
+        @Override
+        public void addTransformer(ClassFileTransformer transformer, boolean canRetransform) {
+        }
+
+        @Override
+        public void addTransformer(ClassFileTransformer transformer) {
+        }
+
+        @Override
+        public boolean removeTransformer(ClassFileTransformer transformer) {
+            return false;
+        }
+
+        @Override
+        public boolean isRetransformClassesSupported() {
+            return false;
+        }
+
+        @Override
+        public void retransformClasses(Class<?>... classes) throws UnmodifiableClassException {
+        }
+
+        @Override
+        public boolean isRedefineClassesSupported() {
+            return false;
+        }
+
+        @Override
+        public void redefineClasses(ClassDefinition... definitions)
+                throws ClassNotFoundException, UnmodifiableClassException {
+        }
+
+        @Override
+        public boolean isModifiableClass(Class<?> type) {
+            return false;
+        }
+
+        @Override
+        public Class<?>[] getAllLoadedClasses() {
+            return new Class<?>[0];
+        }
+
+        @Override
+        public Class<?>[] getInitiatedClasses(ClassLoader loader) {
+            return new Class<?>[0];
+        }
+
+        @Override
+        public long getObjectSize(Object object) {
+            return 0L;
+        }
+
+        @Override
+        public void appendToBootstrapClassLoaderSearch(JarFile jarFile) {
+        }
+
+        @Override
+        public void appendToSystemClassLoaderSearch(JarFile jarFile) {
+        }
+
+        @Override
+        public boolean isNativeMethodPrefixSupported() {
+            return false;
+        }
+
+        @Override
+        public void setNativeMethodPrefix(ClassFileTransformer transformer, String prefix) {
+        }
+
+        @Override
+        public void redefineModule(Module module,
+                                   Set<Module> extraReads,
+                                   Map<String, Set<Module>> extraExports,
+                                   Map<String, Set<Module>> extraOpens,
+                                   Set<Class<?>> extraUses,
+                                   Map<Class<?>, List<Class<?>>> extraProvides) {
+        }
+
+        @Override
+        public boolean isModifiableModule(Module module) {
+            return false;
+        }
+    }
+
+    public static final class RecordingVirtualMachine {
+        private static String attachedProcessId;
+        private static String loadedAgent;
+        private static String loadedAgentArgument;
+        private static boolean detached;
+
+        public static RecordingVirtualMachine attach(String processId) {
+            attachedProcessId = processId;
+            return new RecordingVirtualMachine();
+        }
+
+        public void loadAgent(String agent, String argument) {
+            loadedAgent = agent;
+            loadedAgentArgument = argument;
+        }
+
+        public void detach() {
+            detached = true;
+        }
+    }
+}

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/InstallerTest.java
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/java/net_bytebuddy/byte_buddy_agent/InstallerTest.java
@@ -1,0 +1,167 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package net_bytebuddy.byte_buddy_agent;
+
+import net.bytebuddy.agent.Installer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.lang.instrument.ClassDefinition;
+import java.lang.instrument.ClassFileTransformer;
+import java.lang.instrument.Instrumentation;
+import java.lang.instrument.UnmodifiableClassException;
+import java.security.Permission;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.jar.JarFile;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SuppressWarnings("removal")
+public class InstallerTest {
+    private SecurityManager originalSecurityManager;
+    private boolean securityManagerInstalled;
+
+    @BeforeEach
+    void resetInstaller() {
+        originalSecurityManager = System.getSecurityManager();
+        securityManagerInstalled = false;
+        Installer.premain(null, null);
+    }
+
+    @AfterEach
+    void restoreInstaller() {
+        Installer.premain(null, null);
+        if (securityManagerInstalled && System.getSecurityManager() != originalSecurityManager) {
+            System.setSecurityManager(originalSecurityManager);
+        }
+    }
+
+    @Test
+    void getInstrumentationChecksSecurityManagerPermissionWhenSupported() {
+        RecordingSecurityManager securityManager = new RecordingSecurityManager();
+        Instrumentation instrumentation = new RecordingInstrumentation();
+
+        Installer.premain(null, instrumentation);
+        securityManagerInstalled = installSecurityManagerIfSupported(securityManager);
+
+        assertThat(Installer.getInstrumentation()).isSameAs(instrumentation);
+        if (securityManagerInstalled) {
+            assertThat(securityManager.checkedPermissions)
+                    .extracting(Permission::getName)
+                    .contains("net.bytebuddy.agent.getInstrumentation");
+        } else {
+            assertThat(System.getSecurityManager()).isNull();
+        }
+    }
+
+    private static boolean installSecurityManagerIfSupported(SecurityManager securityManager) {
+        try {
+            System.setSecurityManager(securityManager);
+            return System.getSecurityManager() == securityManager;
+        } catch (UnsupportedOperationException exception) {
+            return false;
+        }
+    }
+
+    private static final class RecordingSecurityManager extends SecurityManager {
+        private final List<Permission> checkedPermissions = new ArrayList<Permission>();
+
+        @Override
+        public void checkPermission(Permission permission) {
+            checkedPermissions.add(permission);
+        }
+    }
+
+    private static final class RecordingInstrumentation implements Instrumentation {
+        @Override
+        public void addTransformer(ClassFileTransformer transformer, boolean canRetransform) {
+        }
+
+        @Override
+        public void addTransformer(ClassFileTransformer transformer) {
+        }
+
+        @Override
+        public boolean removeTransformer(ClassFileTransformer transformer) {
+            return false;
+        }
+
+        @Override
+        public boolean isRetransformClassesSupported() {
+            return false;
+        }
+
+        @Override
+        public void retransformClasses(Class<?>... classes) throws UnmodifiableClassException {
+        }
+
+        @Override
+        public boolean isRedefineClassesSupported() {
+            return false;
+        }
+
+        @Override
+        public void redefineClasses(ClassDefinition... definitions)
+                throws ClassNotFoundException, UnmodifiableClassException {
+        }
+
+        @Override
+        public boolean isModifiableClass(Class<?> type) {
+            return false;
+        }
+
+        @Override
+        public Class<?>[] getAllLoadedClasses() {
+            return new Class<?>[0];
+        }
+
+        @Override
+        public Class<?>[] getInitiatedClasses(ClassLoader loader) {
+            return new Class<?>[0];
+        }
+
+        @Override
+        public long getObjectSize(Object object) {
+            return 0L;
+        }
+
+        @Override
+        public void appendToBootstrapClassLoaderSearch(JarFile jarFile) {
+        }
+
+        @Override
+        public void appendToSystemClassLoaderSearch(JarFile jarFile) {
+        }
+
+        @Override
+        public boolean isNativeMethodPrefixSupported() {
+            return false;
+        }
+
+        @Override
+        public void setNativeMethodPrefix(ClassFileTransformer transformer, String prefix) {
+        }
+
+        @Override
+        public void redefineModule(Module module,
+                                   Set<Module> extraReads,
+                                   Map<String, Set<Module>> extraExports,
+                                   Map<String, Set<Module>> extraOpens,
+                                   Set<Class<?>> extraUses,
+                                   Map<Class<?>, List<Class<?>>> extraProvides) {
+        }
+
+        @Override
+        public boolean isModifiableModule(Module module) {
+            return false;
+        }
+    }
+}

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -1,0 +1,22 @@
+{
+  "reflection" : [
+    {
+      "condition" : {
+        "typeReached" : "net.bytebuddy.agent.ByteBuddyAgent$AttachmentProvider$Accessor$Simple"
+      },
+      "type" : "com.ibm.tools.attach.VirtualMachine"
+    },
+    {
+      "condition" : {
+        "typeReached" : "net.bytebuddy.agent.Attacher"
+      },
+      "type" : "net_bytebuddy.byte_buddy_agent.AttacherTest$RecordingVirtualMachine"
+    },
+    {
+      "condition" : {
+        "typeReached" : "net.bytebuddy.agent.Attacher"
+      },
+      "type" : "net_bytebuddy.byte_buddy_agent.ByteBuddyAgentInnerAgentProviderInnerForByteBuddyAgentTest$RecordingVirtualMachine"
+    }
+  ]
+}

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/resources/META-INF/native-image/reachability-metadata.json
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/src/test/resources/META-INF/native-image/reachability-metadata.json
@@ -2,12 +2,6 @@
   "reflection" : [
     {
       "condition" : {
-        "typeReached" : "net.bytebuddy.agent.ByteBuddyAgent$AttachmentProvider$Accessor$Simple"
-      },
-      "type" : "com.ibm.tools.attach.VirtualMachine"
-    },
-    {
-      "condition" : {
         "typeReached" : "net.bytebuddy.agent.Attacher"
       },
       "type" : "net_bytebuddy.byte_buddy_agent.AttacherTest$RecordingVirtualMachine"

--- a/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/user-code-filter.json
+++ b/tests/src/net.bytebuddy/byte-buddy-agent/1.11.17/user-code-filter.json
@@ -1,0 +1,16 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "net.bytebuddy.agent.**"
+    },
+    {
+      "includeClasses" : "com.ibm.tools.attach.**"
+    },
+    {
+      "includeClasses" : "net_bytebuddy.byte_buddy_agent.**"
+    }
+  ]
+}

